src/docs/theia_ai.md

```diff
@@ -0,0 +1,404 @@
+
+### Variables
+
+Theia AI by default supports two types of variables:
+
+*Global Variables* are available to all agents. Theia and Theia AI already provide a standard set of variables, such as the currently selected text. Tool builders can easily register their own global variables to provide arbitrary additional data from the tool state, including their custom tool components. Global variables can also be used in any prompt template, but are also available to users in the default chat provided by Theia AI, e.g. “#currentText” for the currently selected text.
+
+*Agent-specific Variables* are only available to specific agents (and in their prompt templates), but not to other agents and also not in the default chat.
+
+#### Agent-specific Variables
+
+Agent-specific variables are controlled by a specific agent. To fill a variable with data, you can literally use any internal Theia API or other external APIs or methods you like. As an example, let’s again look at the chat agent that assists users in identifying (and later executing) arbitrary commands in the Theia IDE, such as opening the settings or showing the toolbar. You can review the [full code of this agent](https://github.com/eclipse-theia/theia/blob/master/packages/ai-chat/src/common/command-chat-agents.ts) as a reference.
+
+To allow the underlying LLM to identify the right command, we want to send it a list with all available commands. To achieve this, we first retrieve a list of all available commands using the CommandRegistry (see code example below). This step is independent of Theia AI and you could retrieve any sort of data here instead.
+Second, we resolve the variable ‘command-ids’ when receiving the system prompt’ (also see code example below).
+
+```ts
+// Retrieve all available commands
+const knownCommands: string[] = [];
+for (const command of this.commandRegistry.getAllCommands()) {
+   knownCommands.push(`${command.id}: ${command.label}`);
+}
+
+// Resolve the variable when retrieving the system prompt
+const systemPrompt = await this.promptService.getPrompt('command-chat-agent-system-prompt-template', {
+   'command-ids': knownCommands.join('\n')
+});
+```
+
+Please note that the manual resolvement of variables is not necessary when using global variables.
+Now we can simply use our agent-specific variable in our prompt templates, e.g. like this:
+
+```md
+You are a service that helps users find commands to execute in an IDE.
+
+
+These are the available commands:
+Begin List:
+{{command-ids}}
+End List
+
+
```

Comment:
```suggestion
```md
You are a service that helps users find commands to execute in an IDE.

These are the available commands:
Begin List:
{{command-ids}}
End List
```